mw-webapp/src/utils/auth/handleAuthState.ts

```diff
@@ -0,0 +1,12 @@
+import {User, onAuthStateChanged} from "firebase/auth";
+import {auth} from "src/firebase";
+
+export const handleUserAuthState = (setUserFunction: React.Dispatch<React.SetStateAction<User | null>>) => {
+  onAuthStateChanged(auth, (currentUser) => {
+    if (currentUser) {
+      setUserFunction(currentUser);
+    } else {
+      setUserFunction(null);
+    }
```

Comment:
the same as setUserFunction(currentUser ?? null),

Reply:
Done, thanks